tests/integration/primary_pool_tests.cpp

```diff
@@ -0,0 +1,192 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "gtest/gtest.h"
+
+#include "camp/camp.hpp"
+
+#include "umpire/config.hpp"
+#include "umpire/ResourceManager.hpp"
+
+#include "umpire/strategy/DynamicPoolList.hpp"
+#include "umpire/strategy/DynamicPoolMap.hpp"
+#include "umpire/strategy/QuickPool.hpp"
+
+#include <random>
+#include <string>
+#include <vector>
+
+template <typename T> struct tag_to_string {};
+
+template<> struct tag_to_string<umpire::strategy::DynamicPoolList> { static constexpr const char* value = "DynamicPoolList"; };
+template<> struct tag_to_string<umpire::strategy::DynamicPoolMap>  { static constexpr const char* value = "DynamicPoolMap"; };
+template<> struct tag_to_string<umpire::strategy::QuickPool>       { static constexpr const char* value = "QuickPool"; };
+
+struct host_resource_tag {};    template<> struct tag_to_string<host_resource_tag>      { static constexpr const char* value = "HOST"; };
```

Comment:
Please use newlines 🤦

Reply:
done

Reply:
If you agree, would you mark conversation as reolved?  Thanks.